junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -172,6 +175,76 @@ private static int getOrder(MethodDescriptor descriptor) {
 		}
 	}
 
+	/**
+	 * {@code MethodOrderer} that sorts methods based on the {@link Depend @Depend}
+	 * annotation.
+	 */
+	class DependAnnotation implements MethodOrderer {
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		private static Map<String, Integer> dependencySize;
+		private static Map<String, String[]> digraph;
+
+		@Override
+		public void orderMethods(MethodOrdererContext context) {
+			digraph = context.getMethodDescriptors().stream().filter(
+				descriptor -> descriptor.findAnnotation(Depend.class).isPresent()).collect(
+					Collectors.toMap(descriptor -> descriptor.getMethod().getName(),
+						descriptor -> descriptor.findAnnotation(Depend.class).map(Depend::methods).get()));
+
+			dependencySize = new HashMap<>();
+
+			try {
+				for (String name : digraph.keySet()) {
+					if (!dependencySize.containsKey(name)) {
+						int sz = DFS(name);
+						if (sz == -1) {
+							throw new IllegalArgumentException("Cycle Detected!!!");
+						}
+					}
+				}
+			}
+			catch (IllegalArgumentException exception) {
+				logger.error(exception,
+					() -> "ERROR - Some arguments from @Depend annotations form cyclic dependencies, which would cause undefined behavior!");
+			}
+
+			context.getMethodDescriptors().sort(Comparator.comparing(DependAnnotation::getDependencySize));
+		}
+
+		private static int getDependencySize(MethodDescriptor descriptor) {
+			return dependencySize.getOrDefault(descriptor.getMethod().getName(), 0);
+		}
+
+		private static int DFS(String name) {
```

Comment:
Assuming you mean depth-first search by "DFS", please rename this to `depthFirstSearch()` and document what it does in any case.

Reply:
Done renaming and commenting it usage.